File: Scala/src/main/scala/Player/Player.scala
package Player

class Player() {
  private val PLAYER_HP = 500
  private val PLAYER_DEFENSE = 30
  private var _hp: Int = PLAYER_HP
  private val _defense: Int = PLAYER_DEFENSE

  def Hp: Int = _hp
  def Hp_= (delta: Int): Unit = {
    _hp = delta
  }

  def Defense: Int = _defense
}

File: rust/src/shared/structs/player.rs
pub struct Player {
    pub hp: i32,
    pub defense: i32,
}

impl Player {
    const PLAYER_HP: i32 = 500;
    const PLAYER_DEFENSE: i32 = 30;

    pub fn new() -> Self {
        Player {
            hp: Player::PLAYER_HP,
            defense: Player::PLAYER_DEFENSE,
        }
    }
}
